README.md

```diff
@@ -18,226 +20,374 @@ _For more information see the [TC39 proposal process](https://tc39.github.io/pro
 a new function. Partial application is supported after a fashion in ECMAScript today through the 
-use of either `Function#bind` or arrow functions:
+use of either `Function.prototype.bind` or arrow functions:
 
 ```js
 function add(x, y) { return x + y; }
 
-// Function#bind
+// Function.prototype.bind
 const addOne = add.bind(null, 1);
 addOne(2); // 3
 
 // arrow functions
 const addTen = x => add(x, 10);
 addTen(2); // 12
-
-// arrow functions and pipeline
-const newScore = player.score
-  |> _ => add(7, _)
-  |> _ => clamp(0, 100, _); // deeply nested stack, the pipe to `clamp` is *inside* the previous arrow function.
 ```
 
 However, there are several of limitations with these approaches:
 
-* `bind` can only fix the leading arguments of a function.
-* `bind` requires you explicitly specify the `this` receiver.
-* You cannot easily use `bind` with a template expression or tagged template expression.
-* Arrow functions can be cumbersome when paired with the [pipeline proposal](https://github.com/gilbert/es-pipeline-operator):
-  * Need to write `|> _ =>` for each step in the pipeline.
-  * Unclear as to which stack frame we are in for the call to `clamp`. This can affect available stack space and 
-    is harder to debug.
+* `Function.prototype.bind` can only fix the leading arguments of a function.
+* `Function.prototype.bind` requires you explicitly specify the `this` receiver.
+* Arrow functions lazily re-evaluate their bodies, which can introduce unintended side-effects.
+
+To resolve these concerns, we propose the introduction of the following new language features:
 
-To resolve these concerns, we propose leveraging the `?` token to act as an "argument placeholder" 
-for a non-fixed argument:
+- A new calling convention, `~()`, to indicate a _partial application_ of a call or `new` expression which
+  results in a _partially applied function_.
+- Using the `?` token to act as a _placeholder argument_ for any non-fixed argument in a _partial application_.
+- Using the `?` token followed by a decimal integer literal (i.e., `?0`) to act as an _ordinal
+  placeholder argument_ for non-fixed arguments bound to a specific ordinal parameter in the resulting _partially
+  applied function_.
+- Using the `...` token to act as a _rest placeholder argument_ for any excess arguments.
 
 ```js
-const addOne = add(1, ?); // apply from the left
+const add = (x, y) => x + y;
+const identity = x => x;
+
+// apply from the left:
+const addOne = add~(1, ?);
 addOne(2); // 3
 
-const addTen = add(?, 10); // apply from the right
+// apply from the right:
+const addTen = add~(?, 10);
 addTen(2); // 12
 
-// with pipeline
-let newScore = player.score
-  |> add(7, ?)
-  |> clamp(0, 100, ?); // shallow stack, the pipe to `clamp` is the same frame as the pipe to `add`.
+// accept a fixed argument list:
+const numbers = ["1", "2", "3"].map(parseInt~(?, 10)); // [1, 2, 3]
```

Comment:
I love this example.  "One way to fix a problem you probably didn't know was plaguing you."